### DISCUSSION TITLE
'Palette' does not have a property 'color-scheme'

### DISCUSSION THREAD

**Discussion by realroot2185** [slint-ui/slint#6311] [Q&A] [answered]
I am trying the example https://github.com/slint-ui/slint/blob/master/examples/gallery/gallery.slint.
This file is https://github.com/slint-ui/slint/blob/master/examples/gallery/ui/pages/page.slint.
I see that ColorScheme is built-in enum https://releases.slint.dev/1.7.2/docs/slint/src/language/builtins/enums#colorscheme.
```
import { Switch, GridBox, ListView, ScrollView, HorizontalBox, VerticalBox, Palette } from "std-widgets.slint";

export component Page inherits VerticalBox {
    in property<string> title: "title";
   ...
    HorizontalBox {
    ...
        Switch {
            horizontal-stretch: 0;
            text: @tr("Dark Mode");
            checked: Palette.color-scheme == ColorScheme.dark; // Error here
            ...
```
I have error: `'Palette' does not have a property 'color-scheme'`.
Do I need to declare the dark and light `ColorSheme`s?

**Comment by ogoffart** [comment]
What version of Slint are you using? Could it be that it is an outdated version?
What platform are you on? Did you select a specific style?

**Reply by realroot2185** [reply]
Yes I did not check the version sorry, now updated to v1.8.0.

`OS: Artix Linux x86_64`
I did not select a style yet.
Therefore I am using qt style https://releases.slint.dev/1.8.0/docs/slint/src/advanced/style.html.

Can you specify a `ColorScheme.dark` and `ColorScheme.light` for qt?
I see that there is not something built-in.
So that one can use dark, light and system qt theme.
Is possible to do that for cosmic style?
Adding a custom theme alongside the dark and light ones.

I have a custom kvantum theme with black and:
```
Rectangle {
    background: Palette.background.brighter(60%);
```
does not work.
It works fine with system themes and if I use `background: Palette.accent-background.brighter(60%);` with the custom theme.
So it seems a problem with the black color.

I have this project in Qt Creator and it still shows these warnings (but no error anymore):
```
'Palette' does not have a property 'color-scheme'
Cannot access id 'ColorScheme'
```

**Reply by ogoffart** [reply, marked as answer]
I believe you're still using an old bersion of the slint language server. (slint-lsp)
What version do you see if you preview `AboutSlint {}`

**Reply by realroot2185** [reply]
Correct, updating it fixed it.
I am now checking the rest with https://crates.io/crates/cargo-update.
